### DISCUSSION TITLE
Allow contract address overriding in generated hooks dynamically

### DISCUSSION THREAD

**Discussion by tractorss** [wevm/wagmi#4241] [Ideas]
Currently, there is no option to config the address in the generated hooks dynamically, we can only configure them in [wagmi.config.ts](https://wagmi.sh/cli/config/options) .

### Rationale

There are instances where an app might have a Factory contract that deploy other contracts (Factory contracts have an implementation contract to deploy the new ones , by creating an instance of the implementation contract ).

The hooks for the implementation contract only has the address of the implementation contract, but we need to interact with the other deployed contracts.

Since we do not have the addresses of the deployed contracts before build we cannot configure them.

So it would be ideal to provide dynamic contract address ( or override) without having to use `createUseSimulateContract` or `createUseWriteContract` in the app to create a hook separately for the dynamic address which would render the hooks generated through `generate` useless as we will be creating a new hook just to change the contract address.

### Example

if there is a generated hook like this :
```
const { data } = useSimulateChildContractFunctionA({
   args : [...]
})
```

we should be able to provide a different address for ChildContract like this :
```
const { data } = useSimulateChildContractFunctionA({
    address: differentAddress // defaults to the contract used to generate hook
    args : [...]
})
```

The same goes for Read hooks too.

> In Wagmi/cli v1 , we were able to override the address field , even tho typescript threw errors, in wagmi/cli v2 it's not possible anymore

@tmm

**Comment by tmm** [comment]
Cross-posting [workaround from Discord](https://discord.com/channels/1156791276818157609/1281251545631035463/1281286333096132713): In the meantime, you can build your own hooks with the proxy address using the `'wagmi/codegen'` entrypoint `create*` utils and the ABI output from Wagmi CLI.

```ts
import { createUseReadContract, createUseWriteContract } from 'wagmi/codegen'
import { yourContractAbi } from './generated' // @wagmi/cli output

const proxyAddress = '0x...'
export const useReadYourContract = createUseReadContract({ abi: yourContractAbi, address: proxyAddress })
export const useWriteYourContract = createUseWriteContract({ abi: yourContractAbi, address: proxyAddress })
```

`'wagmi/codegen'` entrypoint also exports `createUseSimulateContract` and `createUseWatchContractEvent`.
